ote/src/clj/ote/services/pre_notices/operator.clj

```diff
@@ -43,29 +44,43 @@
                  (specql/columns ::transit/pre-notice)
                  {::t-operator/id (op/in (authorization/user-transport-operators db user))})))
 
-(defn save-pre-notice [db user notice]
-  (authorization/with-transport-operator-check
-    db user (::t-operator/id notice)
-    (fn []
-      (tx/with-transaction db
-         (let [form-attachments (:attachments notice)
-               n (-> notice
-                     (dissoc :attachments)
-                     (dissoc ::transit/attachments)
-                     (modification/with-modification-fields ::transit/id user))
-               saved-notice (specql/upsert! db ::transit/pre-notice n)
-               notice-id (::transit/id saved-notice)]
-           (log/debug "Save notice: " saved-notice)
-           (specql/update! db ::transit/pre-notice-attachment
-                           {::transit/pre-notice-id notice-id}
-                           (op/and
+(defn- list-attachments-in-db [db user]
+  (specql/fetch db ::transit/pre-notice-attachment
+                (specql/columns ::transit/pre-notice-attachment)
+                {::modification/created-by (authorization/user-id user)}))
+
+(defn- delete-attachments [db config attachments]
+  (doseq [{id ::transit/id} attachments]
+    (specql/delete! db ::transit/pre-notice-attachment {::transit/id id}))
+  (attachment/delete-from-aws db config attachments))
```

Comment:
What happens if the deletion from s3 bucket fails for some reason? 
Pre-notice-attachments are already removed at this point, so we cannot retry the s3 delete query in case of failure. However, I don't think it is a  big issue if there are some orphaned files in the bucket. They can be addressed with s3 lifecycle management.

Reply:
S3 call is inside the transaction and failure should rollback the transaction.